.github/workflows/build_and_run.yaml

```diff
@@ -88,13 +88,27 @@ jobs:
       - name: Run training
         run: |
           poetry --ansi run poe run_training --config ../workflow/config_for_tests.yml
-  
+
+      - name: Dump logs
+        if: success() || failure()
+        uses: actions/upload-artifact@v4
+        with:
+          name: logs
+          path: ./**/*.log
+
+      - name: Dump study models
+        if: success() || failure()
+        uses: actions/upload-artifact@v4
+        with:
+          name: models
+          path: ./**/*.pkl
+
       - name: Sanity check
         working-directory: ./AIAgent
         run: |
-            if ls report/study.pkl 1> /dev/null 2>&1; then
-                echo "PKL files found."
-            else
-                echo "No PKL files found."
-                exit 1
-            fi
+          if ls report/study.pkl 1> /dev/null 2>&1; then
```

Comment:
Can it be replaced with [this one](https://github.com/actions/upload-artifact?tab=readme-ov-file#customization-if-no-files-are-found)?